keps/sig-network/20201014-NamespaceAsName.md

```diff
@@ -0,0 +1,332 @@
+bogged down.
+-->
+
+#### Story 1
+
+As an administrator i want to allow access from monitoring pods in kube-system namespace, into any pod in my cluster, as a fundamental policy. 
+
+#### Story 2
+
+As an user I want to "just add" a namespace to my allow list without having to manage the labels which might get added/removed over time from said namespace.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+- CNIs may not choose, initially, to support this construct, and diligent communication with CNI providers will be needed to make sure it's widely adopted.
+- CNIs may not choose, initially to support this construct, and dilligent communication with CNI providers will be needed to make sure its widely adopted.
+- We need to make sure that hidden defaults don't break the meaning of existing policys, for example:
+  - if a namespcee name is retrieved by an api client which doesn't yet support it, the client doesnt crash (and the plugin doesnt crash, either).
+  - if a namespace name policy doesn't exist (is null), then the policy should behave identically to any policy made before this field was added, that is, the `NetworkPolicyPeer` should be as obvious as possible to interpret in a way that is compatible with the semantics of the v1 policy API **before** this feature was added to it.  
+
+## Design Details
+
+- Add a new selector to the network policy peer data structure which can switch between allowing a "matchName" or "matchLabels" implementation, supporting a policy that is expressed like this:
+
+- A list of conventional namespaces
+The "conventional namespaces" will look like so:
+```
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: mysql-allow-app
+spec:
+  podSelector:
+    matchLabels:
+      app: mysql
+  ingress:
+  - from:
+    - namespaceSelector:
+         matchName:
+         -  my-frontend
+```
+
+### Test Plan
+
+We will add tests for this new api semantic into the exting test/e2e/ network policy test suites in upstream which cover both of these scenarios, using the validation framework outlined in https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20200204-NetworkPolicy-verification-rearchitecture.md#motivation 
+
+### Graduation Criteria
+
+Note that we may not have explicity feature gates for this KEP, because NetworkPolicies typically haven't been implemented with gates.  We keep this section for the time being in any case as it organizes our thinking around the general progression of the feature over time.
+
+#### Alpha 
+- Communicate CNI providers about the new field
+- Add validation tests in API which confirm several positive / negative scenarios in the test matrix for when this field is present/absent
+
+#### Beta
+- The name selector has been supported for at least 1 minor release
+- Four commonly used CNI providers implement the new field, with generally positive feedback on its usage.
```

Comment:
Why is this written in terms of CNI providers instead of NetPol providers?

Reply:
Updated